cli/src/remote/watcher.js

```diff
@@ -82,45 +86,65 @@ export default class RemoteWatcher {
   }
 
   // Pull multiple files/dirs metadata at once, given their ids
-  async pullMany (ids: string[]) {
-    let failedIds = []
+  async pullMany (changes: Array<{id: string, deleted?: true}>) {
+    let failedChanges = []
 
-    for (let id of ids) {
+    for (const change of changes) {
       try {
-        await this.pullOne(id)
+        await this.pullOne(change)
       } catch (err) {
         log.error({err})
-        failedIds.push(id)
+        failedChanges.push(change)
       }
     }
 
-    if (failedIds.length > 0) {
+    if (failedChanges.length > 0) {
       throw new Error(
-        `Some documents could not be pulled: ${failedIds.join(', ')}`
+        `Some changes could not be pulled:\n${failedChanges.map(c => JSON.stringify(c)).join('\n')}`
       )
     }
   }
 
   // Pull a single file/dir metadata, given its id
-  async pullOne (id: string): Promise<*> {
-    const doc: ?RemoteDoc = await this.remoteCozy.findMaybe(id)
-
-    if (doc != null) {
-      return this.onChange(doc)
+  async pullOne (change: {id: string, deleted?: true}): Promise<*> {
+    const {id, deleted} = change
+    let doc
+    if (!deleted) doc = await this.remoteCozy.findMaybe(id)
```

Comment:
@aenario we don't allow the `include_docs` parameter for the changes feed in cozy-stack. Do you see a reason for that? Else, I think it would be simpler to use it here.